### PR TITLE
add aten::pixel_shuffle implementation (#6328)

### PR DESCRIPTION
**Pull request by wjliu1998**
Following the implementation of aten::pixel_shuffle in pytorch. The implementation of pixel_shuffle has been added in pytorch frontend

### CODE REVIEW

**Comment by masahi** [comment]
thanks, please add a test.

**Comment by wjliu1998** [comment]
> thanks, please add a test.

Thanks! The test has been added!

File: python/tvm/relay/frontend/pytorch.py [hunk 1]
```diff
@@ -1230,6 +1230,42 @@ def _impl(inputs, input_types):
...
+        num_inputs = len(inputs)
+        oc = c // upscale_squared
+        oh = h * upscale_factor
+        ow = w * upscale_factor
+
+        new_shape = [b, oc, upscale_factor, upscale_factor, h, w]
+        out_shape = [b, oc, oh, ow]
+
+        data = _op.transform.reshape(data, new_shape)
+        axes = [0, 1, 4, 2, 5, 3]
```
Comment (masahi) [inline python/tvm/relay/frontend/pytorch.py:1265]: Can you tell me what are these hardcoded axes?

Comment (wjliu1998) [reply python/tvm/relay/frontend/pytorch.py:1265]: The original shape order is [b, oc, upscale_factor, upscale_factor, h, w], the hardcoded axes transpose the order to [b, oc, h, upscale_factor, w, upscale_factor] for further reshape

Comment (masahi) [reply python/tvm/relay/frontend/pytorch.py:1265]: ok, please add your explanation above as a comment (so that people won't get confused).

Comment (wjliu1998) [reply python/tvm/relay/frontend/pytorch.py:1268]: The comment has been added.

**Comment by masahi** [comment]
@wjliu1998 lint is complaining

File: python/tvm/relay/frontend/pytorch.py [hunk 2]
```diff
@@ -1230,6 +1230,45 @@ def _impl(inputs, input_types):
...
+        elif isinstance(data, list):
+            ndims = data
+        elif isinstance(data, (torch.Tensor, np.ndarray)):
+            ndims = _infer_shape(data)
+        else:
+            msg = "Data type %s could not be parsed in transpose op" % (type(data))
+            raise AssertionError(msg)
+
+        if isinstance(data, tvm.runtime.NDArray):
+            ndims = len(_infer_shape(data))
```
Comment (masahi) [inline python/tvm/relay/frontend/pytorch.py:1254]: It seems you copy pasted the above code from `_transpose()`. That one is a legacy code we don't want to repeat. Please leave only what is necessary.

Comment (wjliu1998) [reply python/tvm/relay/frontend/pytorch.py]: Thanks! The code has been deleted

Comment (masahi) [reply python/tvm/relay/frontend/pytorch.py]: I mean, you have three `if isinstance(...)` right? You shouldn't need `isintance`, just get ndims directly (figure out which code path actually hit in your test).

Comment (wjliu1998) [reply python/tvm/relay/frontend/pytorch.py]: Thank you for pointing out that. It seems `ndims = len(_infer_shape(data, prelude.mod))` is enough

**Review by masahi — APPROVED** [review]

**Comment by masahi** [comment]
Thanks @wjliu1998

**Pull request merged**